wallet/api/client_check_transaction.go

```diff
@@ -0,0 +1,198 @@
+	}
+
+	w, err := h.walletStore.GetWallet(ctx, connectedWallet.Name())
+	if err != nil {
+		if errors.Is(err, ErrWalletIsLocked) {
+			h.interactor.NotifyError(ctx, traceID, ApplicationError, err)
+		} else {
+			h.interactor.NotifyError(ctx, traceID, InternalError, fmt.Errorf("could not retrieve the wallet associated to the connection: %w", err))
+		}
+		return nil, internalError(ErrCouldNotCheckTransaction)
+	}
+
+	request.PubKey = params.PublicKey
+	if errs := wcommands.CheckSubmitTransactionRequest(request); !errs.Empty() {
+		return nil, invalidParams(errs)
+	}
+
+	if err := h.interactor.NotifyInteractionSessionBegan(ctx, traceID); err != nil {
+		return nil, requestNotPermittedError(err)
+	}
+	defer h.interactor.NotifyInteractionSessionEnded(ctx, traceID)
+
+	if connectedWallet.RequireInteraction() {
+		receivedAt := time.Now()
+		approved, err := h.interactor.RequestTransactionReviewForChecking(ctx, traceID, connectedWallet.Hostname(), connectedWallet.Name(), params.PublicKey, params.RawTransaction, receivedAt)
+		if err != nil {
+			if errDetails := handleRequestFlowError(ctx, traceID, h.interactor, err); errDetails != nil {
+				return nil, errDetails
+			}
+			h.interactor.NotifyError(ctx, traceID, InternalError, fmt.Errorf("requesting the transaction review failed: %w", err))
+			return nil, internalError(ErrCouldNotCheckTransaction)
+		}
+		if !approved {
+			return nil, userRejectionError(ErrUserRejectedCheckingOfTransaction)
+		}
+	}
+
+	h.interactor.Log(ctx, traceID, InfoLog, "Looking for a healthy node...")
+	currentNode, err := h.nodeSelector.Node(ctx, func(reportType node.ReportType, msg string) {
+		h.interactor.Log(ctx, traceID, LogType(reportType), msg)
+	})
+	if err != nil {
+		h.interactor.NotifyError(ctx, traceID, NetworkError, fmt.Errorf("could not find a healthy node: %w", err))
+		return nil, nodeCommunicationError(ErrNoHealthyNodeAvailable)
+	}
+
+	h.interactor.Log(ctx, traceID, InfoLog, "Retrieving latest block information...")
+	stats, err := currentNode.SpamStatistics(ctx, request.PubKey)
+	if err != nil {
+		h.interactor.NotifyError(ctx, traceID, NetworkError, fmt.Errorf("could not get the latest block from the node: %w", err))
+		return nil, nodeCommunicationError(ErrCouldNotGetLastBlockInformation)
+	}
+	h.interactor.Log(ctx, traceID, SuccessLog, "Latest block information has been retrieved.")
+
+	if stats.LastBlockHeight == 0 {
+		h.interactor.NotifyError(ctx, traceID, NetworkError, ErrCouldNotGetLastBlockInformation)
+		return nil, nodeCommunicationError(ErrCouldNotGetLastBlockInformation)
+	}
+
+	if stats.ChainID == "" {
+		h.interactor.NotifyError(ctx, traceID, NetworkError, ErrCouldNotGetChainIDFromNode)
+		return nil, nodeCommunicationError(ErrCouldNotGetChainIDFromNode)
+	}
+
+	err = h.spam.CheckSubmission(request, &stats)
+	if err != nil {
+		h.interactor.NotifyError(ctx, traceID, ApplicationError, fmt.Errorf("could not send transaction: %w", err))
```

Comment:
```suggestion
		h.interactor.NotifyError(ctx, traceID, ApplicationError, fmt.Errorf("could not send transaction: %w", err))
```

Should "err" be error here?

Reply:
`err` here is a variable holding the error
it is set on line 123